lib/packages.js

```diff
@@ -379,6 +379,8 @@ function _sortPackages(cache) {
   const unsorted = Object.keys(cache).map(name => cache[name]);
   const sorted = [];
 
+  _detectCycle(cache);
```

Comment:
Should expose this function as a "private" function so it can be tested, i.e. `api._detectCycle = _detectCycle`. Then we should have both positive and negative tests with some simple mock manifests. It's a little difficult to tell if it's going to do the right thing when a number of dependencies are shared but there is actually no cycle. Explicit tests would make this easier and prevent regressions.